### PR TITLE
Centralizing currentAuthor logic.

### PR DESCRIPTION
**Pull request by derekdowling**
This basically makes currentAuthor a prop for anything that needs it and is handled getting set at the highest component level which is the Layout.jsx comp. A bit of hackery ensued in order to make this work with React-Router. The good news is they have fixes coming for some of the problems. This also gives us nice entry points for doing checkAuth calls and routing the user to login if they aren't currently authed.

### CODE REVIEW

File: app/scripts/components/layout.jsx [hunk 1]
```diff
@@ -1,19 +1,52 @@
...
+    // we do this so we can pass essentially a global prop into the app in the
+    // form of the currently logged in user
     return (
       <Grid fluid={true}>
-        <Navbar />
+        <Navbar author={this.state.currentAuthor} />
         <Grid>
           <Col md={8} mdOffset={2}>
-            <RouteHandler />
+            <AppHandler currentAuthor={this.state.currentAuthor} />
```
Comment (derekdowling) [inline app/scripts/components/layout.jsx]: @KarimBaaba if I get this working, then we don't need to do componentDidMount everywhere to get the current user and we don't need to do upstream calls to the AuthorStore to get it either.

File: app/scripts/components/timeline.jsx [hunk 2]
```diff
@@ -1,54 +1,55 @@
...
   // If a user logs out and causes a state change within
   // The current page then make sure render() doesn't update.
   // A transition will eventually occure...
-  shouldComponentUpdate: function(nextProps, nextState) {
-    if (Check.undefined(nextState.currentAuthor)) {
-      return false;
-    }
-    return true;
-  },
+  // shouldComponentUpdate: function(nextProps, nextState) {
```
Comment (derekdowling) [inline app/scripts/components/timeline.jsx]: Commented this out as not to nuke all the old logic if I fail.

**Comment by derekdowling** [comment]
Hmm, I tried rebasing a couple of different ways here and ended up in a shitty state. I'm just going to leave it.

File: app/scripts/components/navbar.jsx [hunk 3]
```diff
@@ -5,27 +5,21 @@ import { NavItemLink } from 'react-router-bootstrap';
...
-    return {
-      currentAuthor: AuthorStore.getCurrentAuthor()
-    };
-  },
+  mixins: [Navigation],
 
   logOut: function(evt) {
     evt.preventDefault();
-    AuthorStore.logOut();
+    AuthorActions.logOut();
```
Comment (xkb) [inline app/scripts/components/navbar.jsx:17]: `AuthorActions.logout();`

File: app/scripts/stores/author.js [hunk 4]
```diff
@@ -31,11 +32,15 @@ var ALIST = [
...
 
-  authorList: ALIST,
+    // loggedIn is hack for dealing with a problem in react router
+    this.loggedIn = false;
+    this.currentAuthor = FIXTURE;
+    this.authorList = ALIST;
 
-  init: function() {
+    this.listenTo(AuthorActions.checkAuth, this.checkAuth);
+    this.listenTo(AuthorActions.login, this.logIn);
```
Comment (xkb) [inline app/scripts/stores/author.js:43]: This is no longer needed. Only complete and failed will fire.

**Pull request merged**